c/tskit/tables.c

```diff
@@ -7939,6 +7939,42 @@ tsk_table_collection_check_migration_integrity(
     return ret;
 }
 
+static int TSK_WARN_UNUSED
+tsk_table_collection_check_individual_integrity(
+    const tsk_table_collection_t *self, tsk_flags_t options)
+{
+    int ret = 0;
+    tsk_size_t j, k;
+    const tsk_individual_table_t individuals = self->individuals;
+    const tsk_id_t num_individuals = (tsk_id_t) individuals.num_rows;
+    const bool check_individual_ordering = options & TSK_CHECK_INDIVIDUAL_ORDERING;
+
+    /* Check parent references are valid */
+    for (j = 0; j < individuals.parents_length; j++) {
+        if (individuals.parents[j] != TSK_NULL
+            && (individuals.parents[j] < 0
+                   || individuals.parents[j] >= num_individuals)) {
+            ret = TSK_ERR_INDIVIDUAL_OUT_OF_BOUNDS;
+            goto out;
+        }
+    }
+
+    /* Check parents are ordered */
+    if (check_individual_ordering) {
+        for (j = 0; j < (tsk_size_t) num_individuals; j++) {
+            for (k = individuals.parents_offset[j];
+                 k < individuals.parents_offset[j + 1]; k++) {
+                if (individuals.parents[k] != TSK_NULL
+                    && individuals.parents[k] >= (tsk_id_t) j) {
+                    ret = TSK_ERR_UNSORTED_INDIVIDUALS;
```

Comment:
I suppose we might check that no individual is its own parent (even if we are not checking individual ordering).

Reply:
Could be a useful diagnostic. No good reason to be your own parent (unless people are being sloppy with the tables and want to fix things up at the end).

Reply:
I'll add a test for this, as this code already forbids it.